src/tox/config/__init__.py

```diff
@@ -1498,9 +1498,10 @@ def _getenvdata(self, reader, config):
 
         env_list = []
         envlist_explicit = False
-        if (from_option and "ALL" in from_option) or (
-            not from_option and from_environ and "ALL" in from_environ.split(",")
-        ):
+        if (
+            (from_option and "ALL" in from_option)
+            or (not from_option and from_environ and "ALL" in from_environ.split(","))
+        ) and not os.environ.get(PARALLEL_ENV_VAR_KEY_PRIVATE):
```

Comment:
Maybe one just wants to check for

```suggestion
        ) and PARALLEL_ENV_VAR_KEY_PRIVATE not in os.environ:
```